### DISCUSSION TITLE
HmIP-RFUSB not available anymore in docker setup after reboot

### DISCUSSION THREAD

**Discussion by fschrempf** [OpenCCU/OpenCCU#2009] [🇺🇸 Q&A] [answered]
I have a setup where I run the RaspberryMatic docker container inside an LXC vontainer on a Proxmoxx host with the HmIP-RFUSB. After setting things up according to the [docker documentation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Docker-OCI) things worked just fine for several days. Some quirks for the LXC setup are necessary (see [here](https://homematic-forum.de/forum/viewtopic.php?t=67243)) but I doubt that this plays a role in this case.

Now after my host machine got rebooted, suddenly the HmIP-RFUSB is not available anymore. The device is available inside the container, though. This is the first time the whole setup was rebooted. Could it be that the firmware on the HmIP-RFUSB got updated and is now causing these problems?

Thanks for any help getting the setup running again!

```
/ # detect_radio_module /dev/raw-uart
HMIP-RFUSB <redacted> <redacted> 0xFFCEBE 0xBEB208 4.4.18
```

```
/ # cat /var/hm_mode 
HM_HMIP_ADDRESS=''
HM_HMIP_ADDRESS_ACTIVE='0xBEB208'
HM_HMIP_DEV=''
HM_HMIP_DEVNODE=''
HM_HMIP_DEVTYPE='USB'
HM_HMIP_SERIAL=''
HM_HMIP_SGTIN=''
HM_HMIP_VERSION=''
HM_HMRF_ADDRESS=''
HM_HMRF_ADDRESS_ACTIVE='0xFFCEBE'
HM_HMRF_DEV=''
HM_HMRF_DEVNODE=''
HM_HMRF_DEVTYPE=''
HM_HMRF_SERIAL=''
HM_HMRF_VERSION=''
HM_HOST='oci'
HM_LED_GREEN=''
HM_LED_GREEN_MODE1='none'
HM_LED_GREEN_MODE2='heartbeat'
HM_LED_RED=''
HM_LED_RED_MODE1='timer'
HM_LED_RED_MODE2='none'
HM_LED_YELLOW=''
HM_LED_YELLOW_MODE1='none'
HM_LED_YELLOW_MODE2='none'
HM_MODE='NORMAL'
HM_RTC=''
```

```
Identifying host system: oci, OK
Initializing RTC Clock: no hardware found
Running sysctl: OK
Checking for Factory Reset: not required
Checking for Backup Restore: not required
Initializing System: OK
Setup ca-certificates: OK
Starting logging: OK
Populating /dev using udev: starting version 3.2.11
done
Init onboard LEDs: init, OK
Starting irqbalance: OK
Starting iptables: OK
Starting network: eth0: link up, fixed, firewall, inet up, 172.23.0.2, OK
Identifying Homematic RF-Hardware: ...HmRF: n/a, HmIP: n/a, OK
Updating Homematic RF-Hardware: HMIP-RFUSB: =>4.4.18, ERROR ( != 4.4.18)
Starting hs485dLoader: disabled
Starting xinetd: OK
Starting eq3configd: OK
Starting lighttpd: OK
Starting ser2net: disabled
Starting ssdpd: OK
Starting NUT services: disabled
Initializing Third-Party Addons: OK
Starting LGWFirmwareUpdate: ...OK
Setting LAN Gateway keys: OK
Starting hs485d: disabled
Starting multimacd: not required
Starting rfd: no BidCos-RF hardware found
Starting HMIPServer: ....OK
Starting ReGaHss: .OK
Starting CloudMatic: OK
Starting NeoServer: disabled
Starting Third-Party Addons: OK
Starting crond: OK
Setup onboard LEDs: booted, OK
Finished Boot: 3.65.11.20221005 (raspmatic_oci_amd64)
```

**Comment by jens-maus** [comment]
This reads like a problem in the host system of your Proxmox host where you installed the necessary generic raw uart kernel drivers. Could be easily related to you having updated proxmox and its kernel alongside and having missed to reinstall the drivers for the newer kernel.

BTW: I am also working on a direct LXC container-based version of RaspberryMatic which would not require to perform a double virtualization like you did with docker within lxc using proxmox! However, this will still require some time to be finished. In the meantime it would be great if you could provide a detailed step by step documentation on how you installed RaspberryMatic in a LXC container on proxmox as the resource in the homematic forum is only very loosely explaining the process.

**Reply by fschrempf** [reply]
> This reads like a problem in the host system of your Proxmox host where you installed the necessary generic raw uart kernel drivers. Could be easily related to you having updated proxmox and its kernel alongside and having missed to reinstall the drivers for the newer kernel.

Hm, I can't remember changing anything on the host side. I have `/dev/ttyUSB0` and `/dev/raw-uart` on the host, but only the latter in the container. The `/dev/ttyUSB0` is missing in the container. But I guess that's alright?

As far as I can see the drivers are there and reinstalling `pivccu-modules-dkms` doesn't help. Running `ls /sys/class/raw-uart/` returns no result, which seems strange.

> BTW: I am also working on a direct LXC container-based version of RaspberryMatic which would not require to perform a double virtualization like you did with docker within lxc using proxmox! However, this will still require some time to be finished. In the meantime it would be great if you could provide a detailed step by step documentation on how you installed RaspberryMatic in a LXC container on proxmox as the resource in the homematic forum is only very loosely explaining the process.

Sounds interesting. At the moment the two levels of virtualization is not really an issue for me. Also I'm hosting docker containers with other services in the same LXC container.  
I will try to come up with a guide for the LXC setup, but I'm not really sure if I can really provide much more details as already there in the forum post.

**Reply by fschrempf** [reply]
I just found out that there is also the `debmatic` project. Maybe using their LXC setup is the better option for me at the moment. I'm new to all of this so I haven't found out what other differences there are between RaspberryMatic and debmatic.

Still I really don't understand how my perfectly fine setup stopped working because of a reboot.

**Comment by fschrempf** [comment, marked as answer]
Ok, I think I found the problem. Following the guide during the first setup I also created the udev rule in step 6. But the rule probably didn't trigger until I rebooted the system yesterday. The `raw-uart` link created by the rule probably interferes with the generic raw-uart driver. Removing the rule and doing another reboot seems to have fixed the problem.

**Reply by jens-maus** [reply]
Exactly! The problem is/was that you were having the `/etc/udev/rules.d/10-hmiprfusb.rules` installed which is not necessary with recent HmIP-RFUSB firmware versions anymore. Instead you have to make sure only the generic raw uart kernel driver is used and the `cp210x` completely omitted/ignored as it interferes with the generic raw uart! So better would be to add a udev rule that makes sure the `cp210x` driver is NOT loaded at all!
